### DISCUSSION TITLE
Geographical plots

### DISCUSSION THREAD

**Discussion by ibrewster** [pyqtgraph/pyqtgraph#2424] [Ideas]
For many graphing applications, the ability to graph geographical coordinates could be quite helpful. For example, an airline or trucking company may wish to track fleet locations - potentially updated in real-time. Mapping applications may wish to map locations of installations or route tracks. Or, as in my application, you may wish to graph various spatial data types, such as satellite data.

Currently, this is not _easily_ doable in PyQtGraph, though it is possible by converting latitude/longitude data to a projected coordinate system, and writing custom `axisItem` code to convert the projected coordinates back to latitude/longitude for display, as I have done in this screen shot here:
 
<img width="1024" alt="Screen Shot 2022-09-22 at 9 23 11 AM" src="https://user-images.githubusercontent.com/1500095/191811939-6c038aca-2ae8-4fd8-9f0e-ac7332e18325.png">

This is done using the mercator projection, I'm not sure how well this approach would translate to other projections which may not have straight and/or parallel latitude/longitude lines.

 - The AxisItem code used to create the above pictured graph _should_ be accessible at https://bitbucket.org/ijbrewster/tropomi_gui/src/master/TROPOMIQt/LatLonAxisItem.py 
 - The map outline background is created from a SVG that is rendered to a bitmap (for speed of display), then scaled and positioned correctly using (somewhat) magic numbers. 
 - The data is graphed by converting the latitude/longitude corners of the data "pixels" to mercator projection, creating a list of QPainterPath objects, and using those in the PyQtGraph plot command.

**Comment by SpacelySpaceSprockets** [comment]
I second this.  But maybe this would be better done in a separate project that uses pyqtgraph?  In the way that cartopy and basemap are projects that use matplotlib.  Mapping tools tend to be pretty large and this would really increase the size of pyqtgraph I think.

I currently have a project that uses pyqtgraph for plotting, and cartopy for maps, and the requirement to also install matplotlib along with cartopy makes it much larger than it needs to be.
